### PR TITLE
Fix for header parsing when the name has length zero (semicolon on beginning of line)

### PR DESCRIPTION
**Pull request by ewertons**

### CODE REVIEW

File: src/uhttp.c [hunk 1]
```diff
@@ -187,30 +187,39 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
             colonEncountered = true;
             size_t keyLen = (&buffer[index])-targetPos;
```
Comment (danewalton) [inline src/uhttp.c:188]: nit: space this out if you can :)
```suggestion
            size_t keyLen = (&buffer[index]) - targetPos;
```

File: src/uhttp.c [hunk 2]
```diff
@@ -187,30 +187,39 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
...
-                headerKey = NULL;
-            }
-            headerKey = (char*)malloc(keyLen+1);
-            if (headerKey == NULL)
+            if (keyLen == 0)
             {
+                LogError("Invalid header name with zero length.");
                 result = MU_FAILURE;
-                continueProcessing = false;
+                continueProcessing = false;
```
Comment (hihigupt) [inline src/uhttp.c:194]: >                 [](start = 43, length = 16)

remove spaces at the end of line

File: src/uhttp.c [hunk 3]
```diff
@@ -187,30 +187,39 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
...
-                crlfEncounted = false;
+                    // Convert to lower case
+                    for (size_t inner = 0; inner < keyLen; inner++)
+                    {
+                        headerKey[inner] = (char)tolower(headerKey[inner]);
+                    }
+
+                    targetPos = buffer+index+1;
+                    crlfEncounted = false;
+                }
```
Comment (hihigupt) [inline src/uhttp.c:222]: >    [](start = 17, length = 3)

same

File: src/uhttp.c (hunk 2 above)
Comment (ewertons) [reply src/uhttp.c:194]: Done.

File: src/uhttp.c (hunk 3 above)
Comment (ewertons) [reply src/uhttp.c:222]: Done.

File: src/uhttp.c [hunk 4]
```diff
@@ -185,32 +185,41 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
...
-                headerKey[keyLen] = '\0';
-
-                // Convert to lower case
-                for (size_t inner = 0; inner < keyLen; inner++)
+                if (headerKey != NULL)
+                {
+                    free(headerKey);
+                    headerKey = NULL;
+                }
+                headerKey = (char*)malloc(keyLen+1);
```
Comment (ericwol-msft) [inline src/uhttp.c:203]:
```suggestion
                headerKey = (char*)malloc(keyLen + 1);
```

File: src/uhttp.c [hunk 5]
```diff
@@ -185,32 +185,41 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
...
 
-                targetPos = buffer+index+1;
-                crlfEncounted = false;
+                    // Convert to lower case
+                    for (size_t inner = 0; inner < keyLen; inner++)
+                    {
+                        headerKey[inner] = (char)tolower(headerKey[inner]);
+                    }
+
+                    targetPos = buffer+index+1;
```
Comment (ericwol-msft) [inline src/uhttp.c:220]: __
```suggestion
                    targetPos = buffer + index + 1;
```

**Pull request merged**